test/karma/spec/renewToken.js

```diff
@@ -0,0 +1,258 @@
+      expect(oauthUtil.loadFrame).toHaveBeenCalled();
+      expect(e instanceof OauthError).toBe(true);
+      expect(e.message).toBe(error_description);
+    });
+  });
+
+  it('grantType: implicit', function() {
+    return bootstrap({
+      grantType: 'implicit'
+    })
+    .then(() => {
+      sdk.tokenManager.add('accessToken', ACCCESS_TOKEN_PARSED);
+
+      // We are not loading a real iframe
+      spyOn(oauthUtil, 'loadFrame').and.callFake(urlStr => {
+        const url = new URL(urlStr);
+        expect(url.origin).toBe(ISSUER);
+        expect(url.pathname).toBe('/oauth2/v1/authorize');
+        expect(url.searchParams.get('client_id')).toBe(CLIENT_ID);
+        expect(url.searchParams.get('redirect_uri')).toBe(REDIRECT_URI);
+        expect(url.searchParams.get('response_type')).toBe('token');
+        expect(url.searchParams.get('response_mode')).toBe('okta_post_message');
+        expect(url.searchParams.get('scope')).toBe('openid email');
+        expect(url.searchParams.get('state')).toBeTruthy();
+        
+        // nonce will be a random string when renewing access token. Is this expected?
+        // expect(url.searchParams.get('nonce')).toBe(NONCE);
+
+        // Response back to caller
+        const state = url.searchParams.get('state');
+        const scope = url.searchParams.get('scope');
+        var response = {
+          access_token: ACCESS_TOKEN_STR,
+          expires_in: 3600,
+          scope,
+          state,
+          token_type: 'Bearer'
+        };
+
+        // Simulate window.postMessage() from iframe
+        var event = new Event('message');
+        event.data = response;
+        event.origin = ISSUER;
+        window.dispatchEvent(event);
+
+      });
+      return sdk.tokenManager.renew('accessToken');
+    })
+    .then(function(res) {
+      expect(oauthUtil.loadFrame).toHaveBeenCalled();
+      expect(res.accessToken).toBe(ACCESS_TOKEN_STR);
+    });
+  });
+
+  it('grantType: authorization_code', function() {
+    var codeChallenge, codeVerifier;
+
+    return bootstrap({
+      grantType: 'authorization_code'
+    })
+    .then(() => {
+      sdk.tokenManager.add('accessToken', ACCCESS_TOKEN_PARSED);
+
+      mockWellKnown();
+  
+      // We are not loading a real iframe
+      spyOn(oauthUtil, 'loadFrame').and.callFake(urlStr => {
+        const url = new URL(urlStr);
+        expect(url.origin).toBe(ISSUER);
+        expect(url.pathname).toBe('/oauth2/v1/authorize');
+        expect(url.searchParams.get('client_id')).toBe(CLIENT_ID);
+        expect(url.searchParams.get('redirect_uri')).toBe(REDIRECT_URI);
+        expect(url.searchParams.get('response_type')).toBe('code');
+        expect(url.searchParams.get('response_mode')).toBe('okta_post_message');
+        expect(url.searchParams.get('scope')).toBe('openid email');
+        expect(url.searchParams.get('state')).toBeTruthy();
+        
+        // nonce will be a random string when renewing access token. Is this expected?
```

Comment:
Only when one isn't provided in the oauthParams. You can always specify one if you'd like to assert it.